.releaserc.json

```diff
@@ -2,7 +2,15 @@
   "branches": ["main"],
   "preset": "conventionalcommits",
   "plugins": [
-    "@semantic-release/commit-analyzer",
+    [
+      "@semantic-release/commit-analyzer",
+      {
+        "releaseRules": [
+          {"breaking": true, "release": "minor"},
+          {"type": "feat", "release": "patch"}
```

Comment:
Yeah, agree with cpcloud here. Feats should also be minor.

Reply:
My reasoning is that it's more useful to distinguish between breaking and non-breaking than between (potentially breaking) feature and patch.

Turns out (just looked it up), [this is also exactly how cargo treats version compatibility before 1.0.0](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio):

> This compatibility convention is different from SemVer in the way it treats versions before 1.0.0. While SemVer says there is no compatibility before 1.0.0, Cargo considers `0.x.y` to be compatible with `0.x.z`, where `y ≥ z` and `x > 0`.

so there is actually considerable precedence beyond my gut feeling :D

Reply:
Hm, that I get, but how will we introduce a minor version without `feat`?

Reply:
Ah, by adding `BREAKING CHANGE` of course.

Reply:
So, this effectively makes everything a minor release unless there's a `BREAKING CHANGE`. That makes sense to me.

Reply:
Assuming the default rules for the other commit styles never match, that's the idea, yeah.

Reply:
@jvanstraten 's approach is also how NPM addresses this (assuming you use caret ranges which has been the default for a while):

```
Caret ranges are ideal when an author may make breaking changes between 0.2.4 and 0.3.0 releases, which is a common practice. However, it presumes that there will not be breaking changes between 0.2.4 and 0.2.5. It allows for changes that are presumed to be additive (but non-breaking), according to commonly observed practices.
```

So I do prefer @jvanstraten 's approach, it will help us figure out skills/habits for detecting and communicating breaking changes as well.

Reply:
How about leave the SemVer defaults as-is and only tweak the "pre-1.0.0" behavior? I don't know if there's a strong rationale for changing the meaning of `feat` right now.

Reply:
Agreed.  Let's just move forward with `feat` as-is.  This is blocking breaking changes from merging so I'm happy to compromise here and we can adjust as desired.

Reply:
> How about leave the SemVer defaults as-is and only tweak the "pre-1.0.0" behavior?

Don't get me wrong, I fully intend for this to be reverted at 1.0.0, whatever we do, and my logic is not just some arbitrary special case on top of semver (nor does it conflict with it). The way I see it, incrementing the first nonzero digit means breaking or less significant, the second nonzero digit is feature or less significant, and the third is bugfix. Incrementing from 0 to 1 requires discussion and human intervention. The absolute position in the version just says something about how often it can be expected to change, so the number of zero digits effectively communicates the likelihood for a breaking change to occur.

I don't think either approach is any cleaner or more symmetrical than the other (3 digits just isn't enough to see the patterns very clearly), I just think that my/the rust/the npm approach at this stage of development is more useful to users. But I won't die on this hill. I'll change it if I have to.

Reply:
FYI I went ahead and changed it both here and in the docs commit.